pkg/hostagent/port_darwin.go

```diff
@@ -149,8 +151,25 @@ func (plf *pseudoLoopbackForwarder) forward(ac *net.TCPConn) error {
 		return err
 	}
 	defer unixConn.Close()
-	bicopy.Bicopy(ac, unixConn, nil)
-	return nil
+
+	g, _ := errgroup.WithContext(context.Background())
+
+	g.Go(func() error {
+		_, err := io.Copy(unixConn, ac)
+		if errors.Is(err, io.EOF) {
+			err = nil
+		}
+		return errors.Join(err, unixConn.CloseRead(), ac.CloseWrite())
+	})
+	g.Go(func() error {
+		_, err := io.Copy(ac, unixConn)
+		if errors.Is(err, io.EOF) {
+			err = nil
+		}
+		return errors.Join(err, ac.CloseRead(), unixConn.CloseWrite())
+	})
+
+	return g.Wait()
```

Comment:
This will return an error if CloseWrite or CloseRead failed, changing the behavior.

Reply:
This is the same comment in triplicate, so resolving this one.